packages/grid/_modules_/grid/hooks/features/filter/useGridRegisterFilteringMethod.ts

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { GridFilteringMethod, GridFilteringMethodCollection } from './gridFilterState';
+import { GridPreProcessingGroup, useGridRegisterPreProcessor } from '../../core/preProcessing';
+import { GridApiRef } from '../../../models';
+
+export const useGridRegisterFilteringMethod = (
```

Comment:
Small abstraction of the collection update behavior. 
It does not bring a lot of value but without it, the code is `useGridTreeData` / `useGridGroupingColumns` was less readable.